app/policies/activity_policy.rb

```diff
@@ -47,12 +47,8 @@ def resolve
     end
   end
 
-  private def editable_report_for_organisation
-    Report.editable.find_by(organisation: record.organisation)
-  end
-
   private def editable_report_for_organisation_and_fund
-    fund = record.parent.associated_fund
+    fund = record.associated_fund
     Report.editable.find_by(organisation: record.organisation, fund: fund)
```

Comment:
There is a scope on `Report` that implements this so you can write `Report.editable.for_activity(record).first` here.